### DISCUSSION TITLE
How to test a method that transfers AptosCoin to an address?

### DISCUSSION THREAD

**Discussion by dbrdz** [aptos-labs/aptos-developer-discussions#199] [Questions] [answered]
I'm new to aptos and I'm playing around with tokens, coins and transfers. I wrote a method that only transfers 1 apt to another account:

```
public entry fun buy_credits(signer: &signer, dest: &signer) {
        aptos_framework::coin::transfer<AptosCoin>(signer, dest, 1);
 }
```

I'm writing a very simple test to test this method:

```
    #[test(user_one = @0x3, user_two = @0x4)]
    fun test_buy_credits(user_one: signer, user_two: signer) {
        create_account_for_test(signer::address_of(& user_one));
        create_account_for_test(signer::address_of(& user_two));
        buy_credits(&user_one, &user_two)
    }
```

When I execute the test I get this error:

```
assert!(is_account_registered<CoinType>(account_addr), error::not_found(ECOIN_STORE_NOT_PUBLISHED));

Test was not expected to error, but it aborted with code 393221 originating in the module 0000000000000000000000000000000000000000000000000000000000000001::coin rooted here
```

The error code seems to say that the AptosCoin is not initialized in my testing environment. 

Is there an extra step I have to do in order to use AptosCoin transfer on a test?

**Comment by gregnazario** [comment, marked as answer]
In this case, you need to initialize `AptosCoin` for testing.  In the real networks, genesis handles this.

Here's an example:

This code calls a helper function (defined in a different module), that sets up the coin, and accounts.
https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/marketplace/sources/coin_listing.move#L666-L674

This code here sets up the coin and the accounts associated
https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/marketplace/sources/test_utils.move#L20-L50

**Reply by dbrdz** [reply]
Thank you so much! Really appreciate the help!
